### PR TITLE
resolves #974 lock/unlock orchestration

### LINKED ISSUE #974: [Orchestration] Secondary bot must be able to keep control
Ex: During a transactionnel conversation primary bot can detect a "no Orchestration Intent" for a sentence. This sentence may make sense to continue the dialog with the secondary bot...

### PR DESCRIPTION
**Pull request by curinator**
Signed-off-by: Toki RAOSETA <toki.raoseta@gmail.com>

### CODE REVIEW

File: bot/engine/src/main/kotlin/engine/action/ActionMetadata.kt [hunk 1]
```diff
@@ -28,5 +28,6 @@ data class ActionMetadata(
         /** Message which is a reply to another. */
         var replyMessage: ActionReply = ActionReply.UNKNOWN,
         /** Message which contains a quote. */
-        var quoteMessage: ActionQuote = ActionQuote.UNKNOWN
+        var quoteMessage: ActionQuote = ActionQuote.UNKNOWN,
+        var lockOrchestration: Boolean = false
```
Comment (elebescond) [inline bot/engine/src/main/kotlin/engine/action/ActionMetadata.kt:32]: With this attribute we introduce the notion of orchestration in the TOCK core.
Tock should provide a mechanism allowing a third party module to enrich the metadata.

Comment (elebescond) [inline bot/engine/src/main/kotlin/engine/action/ActionMetadata.kt:32]: With this attribute we introduce the notion of orchestration in the TOCK core.
Tock should provide a mechanism allowing a third party module to enrich the metadata

File: bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt [hunk 2]
```diff
@@ -194,3 +203,26 @@ class RunOrchestrationStoryListener(
...
+    get() = userTimeline.userState.getFlag("block_handover_to_secondary_bot")?.toBoolean() ?: false
+    set(value) {
+        userTimeline.userState.setUnlimitedFlag("block_handover_to_secondary_bot", value.toString())
+    }
+
+var BotBus.blockTakeOverFromPrimaryBot: Boolean
+    get() = userTimeline.userState.getFlag("block_take_over_from_primary_bot")?.toBoolean() ?: false
+    set(value) {
+        userTimeline.userState.setUnlimitedFlag("block_take_over_from_primary_bot", value.toString())
+    }
```
Comment (elebescond) [inline bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt:217]: if bot A is primary of bot B and secondary of bot C, the developer must do a blockTakeOverFromPrimaryBot and a blockHandoverToSecondaryBot ?

Comment (elebescond) [reply bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt:217]: Maybe a global method of locking and unlocking function can help developers? ;-)

Comment (curinator) [reply bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt:217]: - Great idea, finding a generic explicit name remains very difficult though

- I think all operations should be converted to functions if we are introducing global functions to ensure consistency(blockTakeOverFromPrimaryBot(), allowTakeOverFromPrimaryBot(), ...)

**Pull request merged**